pyccl/halos/concentration/constant.py

```diff
@@ -11,26 +12,21 @@ class ConcentrationConstant(Concentration):
 
     Args:
         c (float): constant concentration value.
-        mdef (:class:`~pyccl.halos.massdef.MassDef`): a mass
+        mass_def (:class:`~pyccl.halos.massdef.MassDef`): a mass
             definition object that fixes
             the mass definition used by this c(M)
             parametrization. In this case it's arbitrary.
     """
-    __repr_attrs__ = ("mdef", "c",)
+    __repr_attrs__ = ("mass_def", "c",)
     name = 'Constant'
 
-    def __init__(self, c=1, mdef=None):
+    @warn_api(pairs=[("mdef", "mass_def")])
+    def __init__(self, c=1, *, mass_def=MassDef(200, 'critical')):
         self.c = c
-        super(ConcentrationConstant, self).__init__(mdef)
+        super().__init__(mass_def=mass_def)
 
-    def _default_mdef(self):
-        self.mdef = MassDef(200, 'critical')
-
-    def _check_mdef(self, mdef):
+    def _check_mass_def_strict(self, mass_def):
         return False
 
     def _concentration(self, cosmo, M, a):
-        if np.ndim(M) == 0:
-            return self.c
-        else:
-            return self.c * np.ones(M.size)
+        return np.full_like(M, self.c)
```

Comment:
if `M` is a float, this will return something of type `array`.

![image](https://user-images.githubusercontent.com/8809493/229191271-cde52478-d722-4f6d-b5a6-9d2512f71556.png)

Reply:
This is a zero-dimensional array. As far as Python is concerned, it is the same as a float. The only difference is that it is a numpy object.
```python
for i in np.array(0):
    print(i)

TypeError: iteration over a 0-d array

for i in 0:
    print(i)
TypeError: 'int' object is not iterable
```